core/shared/src/main/scala/fs2/interop/flow/StreamSubscriber.scala

```diff
@@ -228,6 +229,10 @@ private[flow] final class StreamSubscriber[F[_], A] private (
       case Complete(canceled) => {
         case Uninitialized(Some(cb)) =>
           Terminal -> run {
+            // We do the updates here,
+            // to ensure they happen after we have secured the state.
+            inOnNextLoop = false
+            buffer = null
```

Comment:
In theory, not needed anymore since we are cancelled.
But, at minimum, this could lead to the `buffer` data to be GC early.